packages/bull-vault/src/AuctionBull.sol

```diff
@@ -136,6 +256,150 @@ contract AuctionBull is UniFlash, Ownable {
         deltaUpper = _deltaUpper;
     }
 
+    /**
+     * @dev hedge function to reduce delta using an array of signed orders
+     * @param _orders list of orders
+     * @param _crabAmount amount of crab to withdraw or deposit
+     * @param _clearingPrice clearing price in weth
+     * @param _wethTargetInEuler target WETH collateral amount in leverage component
+     * @param _wethLimitPrice limit price
+     * @param _isDepositingInCrab true if the rebalance will deposit into Crab, false if withdrawing funds from crab
+     */
+    function fullRebalance(
+        Order[] memory _orders,
+        uint256 _crabAmount,
+        uint256 _clearingPrice,
+        uint256 _wethTargetInEuler,
+        uint256 _wethLimitPrice,
+        uint24 _ethUsdcPoolFee,
+        bool _isDepositingInCrab
+    ) external {
+        require(msg.sender == auctionManager, "AB0");
```

Comment:
I think it is better to separate owner from auction manager, in case we loose access to the auction manager, we can easily replace it. idk, what u think?

Reply:
Also I can only do the changes Sunday night my time

Reply:
just added a setter for ```auctionManager```